swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/pojo/PojoOperationGenerator.java

```diff
@@ -83,9 +83,14 @@ private void wrapParametersToBody(List<ParameterGenerator> bodyFields) {
     bodyModel = new ModelImpl();
     bodyModel.setType(ModelImpl.OBJECT);
     for (ParameterGenerator parameterGenerator : bodyFields) {
-      SwaggerUtils.addDefinitions(swagger, parameterGenerator.getGenericType());
+      parameterGenerator.setHttpParameterType(HttpParameterType.BODY);
+      scanMethodParameter(parameterGenerator);
+
       Property property = ModelConverters.getInstance().readAsProperty(parameterGenerator.getGenericType());
+      property.setDescription(parameterGenerator.getGeneratedParameter().getDescription());
       bodyModel.addProperty(parameterGenerator.getParameterName(), property);
+
+      parameterGenerator.setHttpParameterType(null);
```

Comment:
This code is a bit confusing. Should HttpParameterType set to the original one or it is always BODY?

Reply:
yes
should not keep as "BODY", that will cause multiple bodies
no way to let swagger read ApiParam value, except we copy related logic

i will add some comment for this special logic.